### ISSUE TITLE
Unable to start matrix-coturn.service

### ISSUE THREAD

**Issue by TacticalApples** [spantaleev/matrix-docker-ansible-deploy#2562]
**Playbook Configuration**:

My `vars.yml` file looks like this:

```yaml
---
# The bare domain name which represents your Matrix identity.
# Matrix user ids for your server will be of the form (`@user:<matrix-domain>`).
#
# Note: this playbook does not touch the server referenced here.
# Installation happens on another server ("matrix.<matrix-domain>").
#
# If you've deployed using the wrong domain, you'll have to run the Uninstalling step,
# because you can't change the Domain after deployment.
#
# Example value: example.com
matrix_domain: [REDACTED]

# The Matrix homeserver software to install.
# See:
#  - `roles/custom/matrix-base/defaults/main.yml` for valid options
# - the `docs/configuring-playbook-IMPLEMENTATION_NAME.md` documentation page, if one is available for your implementation choice
matrix_homeserver_implementation: synapse

# A secret used as a base, for generating various other secrets.
# You can put any string here, but generating a strong one is preferred (e.g. `pwgen -s 64 1`).
matrix_homeserver_generic_secret_key: '[REDACTED]'

# By default, the playbook manages its own Traefik (https://doc.traefik.io/traefik/) reverse-proxy server.
# It will retrieve SSL certificates for you on-demand and forward requests to all other components.
# For alternatives, see `docs/configuring-playbook-own-webserver.md`.
matrix_playbook_reverse_proxy_type: playbook-managed-traefik

# This is something which is provided to Let's Encrypt when retrieving SSL certificates for domains.
#
# In case SSL renewal fails at some point, you'll also get an email notification there.
#
# If you decide to use another method for managing SSL certificates (different than the default Let's Encrypt),
# you won't be required to define this variable (see `docs/configuring-playbook-ssl-certificates.md`).
#
# Example value: someone@example.com
devture_traefik_config_certificatesResolvers_acme_email: '[REDACTED]'

# A Postgres password to use for the superuser Postgres user (called `matrix` by default).
#
# The playbook creates additional Postgres users and databases (one for each enabled service)
# using this superuser account.
devture_postgres_connection_password: '[REDACTED]'
```

**Matrix Server:**
 - OS: Ubuntu 20.04
 - Architecture: amd64

**Ansible:**
If your problem appears to be with Ansible, tell us:
- where you run Ansible -- e.g. on the Matrix server itself; on another computer (which OS? distro? standard installation or containerized Ansible?)
- what version of Ansible you're running (see `ansible --version`)

**Problem description**:

I'm trying to install a brand new Matrix server (first time) and I'm running into the error below during the --install-all portion of the ansible playbook. I'm not sure how to address and I've tried searching the web for some guidance with no luck. Anyone know what I need to do to address?
```
ok: [matrix.REDACTED] => (item={'name': 'matrix-traefik-certs-dumper.service', 'priority': 3500, 'groups': ['matrix', 'traefik-certs-dumper']})
failed: [matrix.REDACTED] (item={'name': 'matrix-coturn.service', 'priority': 4000, 'groups': ['matrix', 'coturn']}) => changed=false 
  ansible_loop_var: item
  item:
    groups:
    - matrix
    - coturn
    name: matrix-coturn.service
    priority: 4000
  msg: |-
    Unable to start service matrix-coturn.service: A dependency job for matrix-coturn.service failed. See 'journalctl -xe' for details.
changed: [matrix.REDACTED] => (item={'name': 'matrix-coturn-reload.timer', 'priority': 5000, 'groups': ['matrix', 'coturn']})

PLAY RECAP ***************************************************************************************
matrix.REDACTED : ok=190  changed=4    unreachable=0    failed=1    skipped=337  rescued=0    ignored=0
```

**Additional context**

**Comment by ludoz** [comment]
Same problem here.
Just edited /etc/systemd/system/matrix-coturn.service and commented: 
#Requires=matrix-traefik-certs-dumper-wait-for-domain@matrix.a-donf.com.service
#After=matrix-traefik-certs-dumper-wait-for-domain@matrix.a-donf.com.service

Not sure for "After"
Restart matrix-coturn.service and it's working.

Not sure about the problem.

**Comment by TacticalApples** [comment]
Hmm, yeah I tried this and I still couldn’t get it to work :/

**Comment by bearyjd** [comment]
I seam to be having the same problem -- I think the issue has to do with certificates (LETS ENCRYPT) not getting generated in traefik that is ultimately causing the matrix-coturn dependency failure.

My vars.yml is similar to yours.

**Comment by WowSuchRicky** [comment]
I have the same thing and yes, it appears to be what @bearyjd mentioned. The reason those lines have to be commented out is because those dependency services are not working. 

Will report back with any more findings, I am really curious what is causing this one but the logs for the dumper service don't seem to be super helpful as to why it isn't dumping certs.

**Comment by bearyjd** [comment]
I found this (https://github.com/devture/com.devture.ansible.role.traefik_certs_dumper)

> Then, upon launching your service:

    the "waiter" service will be started as a dependency

    it will wait for certificates for the specified domain (DOMAIN_NAME) to become available (e.g. /devture-traefik-certs-dumper/dumped-certificates/DOMAIN_NAME/certificate.crt and /devture-traefik-certs-dumper/dumped-certificates/DOMAIN_NAME/privatekey.key)

By default, the "waiter" service waits for 30 seconds (configurable via devture_traefik_certs_dumper_waiter_max_iterations) before giving up and aborting execution of your service

> 

in my directory "/matrix/traefik-certs-dumper/dumped-certificates/private" I only have the letsencrypt.key and no SSL certs.

journalctl -xe had no helpful information and I could not find other logs.

**Comment by bearyjd** [comment]
ok... it was a firewall issue for me. port 80 was blocked for incoming connections which prevented incoming connections.  I just opened all ports for the installation and it worked.

**Comment by mwelsch** [comment]
I got the same issue. I tried to switch my setup to play my traefik which I configured for a lot of other services (nextcloud, overleaf, ...):
```
matrix_playbook_reverse_proxy_type: other-traefik-container
matrix_playbook_reverse_proxyable_services_additional_network: web
devture_traefik_certs_dumper_ssl_dir_path: "/path/to/my/traefik"
```
Commented the `Requires` and `After` but it didn't help
After reversing to my old setup, along with `matrix_playbook_reverse_proxy_type: other-on-same-host` my setup worked again. If I can help with testing stuff let me know ;)

**Comment by lukababu** [comment]
I was installing a brand new instance of this and encountered the same error. I double-checked that ports 80, 443, and 8448 were being forwarded and that fixed it for me!

**Comment by scolastico** [comment]
I'm not sure if this solves your problems, but I had the same issue. After some debugging (why the hell does Traefik not log anything to the console like it normally does? I needed to attach to the container and run the entry point manually...), I noticed that Traefik complains that the element (sub-) domain does not have an A or AAAA record. I only set a CNAME like the docs suggested, but this does not seem to be enough.

```log
time="xxx" level=error msg="Unable to obtain ACME certificate for domains \"element.xxx.xx\": unable to generate a certificate for the domains [element.scolati.co]: error: one or more domains had a problem:\n[element.xxx.xx] acme: error: 400 :: urn:ietf:params:acme:error:dns :: DNS problem: NXDOMAIN looking up A for element.xxx.xx - check that a DNS record exists for this domain; DNS problem: NXDOMAIN looking up AAAA for element.xxx.xx - check that a DNS record exists for this domain\n" rule="Host(`element.xxx.xx`)" ACME CA="https://acme-v02.api.letsencrypt.org/directory" providerName=default.acme routerName=matrix-client-element@docker
```

or in short:
```
DNS problem: NXDOMAIN looking up A for element.xxx.xx - check that a DNS record exists for this domain;
DNS problem: NXDOMAIN looking up AAAA for element.xxx.xx - check that a DNS record exists for this domain
```

tl;dr;
Try to set normal A records instead of CNAME and check if it works.

**Comment by spantaleev** [comment]
You likely did not wait enough for DNS records to refresh, which is why `element.DOMAIN` wasn't available.

`CNAME` is not a problem.

**Comment by scolastico** [comment]
I checked the dns record manually, also its cloudflare so its normally available in seconds.

**Comment by bassamanator** [comment]
@scolastico I tried this, changed 


CNAME | element | - | - | - | matrix.domain.com
-- | -- | -- | -- | -- | --


A | element | - | - | - | domainIP
-- | -- | -- | -- | -- | --

but that didn't do anything.

Still stuck on the same error as @TacticalApples.

**Comment by lukababu** [comment]
What's happening between your router and the server? Is it in a VM if so what are you using to virtualize. Is there a proxy server that redirects traffic. What other services are you hosting?

**Comment by bassamanator** [comment]
The matrix server is being deployed on my linode server which does nothing essentially.

I decided to re-image my nanode and now the install worked. There was obviously some sort of conflict.

**Comment by WowSuchRicky** [comment]
For me this issue ended up being a conflict with a local port being open on the machine. I also was running Kubernetes (microk8s) on the same host.  This was preventing the certs from ever getting dumped and thus never working.

I would recommend making sure no ports are being used on the machine other than defaults and if they are, I'd recommend making a VM on the machine and going from there.

**Comment by lukababu** [comment]
Similar for me, I was running Traefik in TrueNAS k8 docker and bypassing that worked out for me. @spantaleev I think this can be closed.

**Comment by SlayerOFWrly** [comment]
> ok... it was a firewall issue for me. port 80 was blocked for incoming connections which prevented incoming connections. I just opened all ports for the installation and it worked.

port 8080 for me, on the router :P
